src/main/java/ua/bondarenkojek/homework/json/TradingShop.java

```diff
@@ -0,0 +1,136 @@
+        ObjectMapper mapper = new ObjectMapper();
+
+        try {
+            mapper.writeValue(file, fruitsDB);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void load(String pathToJsonFile) {
+        File file = new File(pathToJsonFile);
+        ObjectMapper mapper = new ObjectMapper();
+
+        try {
+            fruitsDB = Arrays.asList(mapper.readValue(file, Fruit[].class));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public List<Fruit> getSpoiledFruits(Date date) {
+        return fruitsDB.stream().filter((Fruit fruit) -> {
+            Date fruitDate = fruit.getDate();
+            int shelfLife = fruit.getShelfLife();
+            Calendar calendar = Calendar.getInstance();
+            calendar.setTime(fruitDate);
+            calendar.add(Calendar.DATE, shelfLife);
+            return calendar.getTime().getTime() < date.getTime();
+        }).collect(Collectors.toList());
+    }
+
+    public List<Fruit> getSpoiledFruits(Date date, Fruit.FruitType type) {
+        return getSpoiledFruits(date).stream().filter(fruit -> type == fruit.getType()).collect(Collectors.toList());
+    }
+
+    public List<Fruit> getAvailableFruits(Date date) {
+        List<Fruit> tmp = getSpoiledFruits(date);
+        return fruitsDB.stream().filter(fruit -> !tmp.contains(fruit)).collect(Collectors.toList());
+    }
+
+    public List<Fruit> getAvailableFruits(Date date, Fruit.FruitType type) {
+        return getAvailableFruits(date).stream().filter(fruit -> type == fruit.getType()).collect(Collectors.toList());
+    }
+
+    public List<Fruit> getAddedFruits(Date date) {
+        return fruitsDB.stream()
+                .filter(fruit -> fruit.getDate().getTime() / MILLISECONDS_FROM_DAY == date.getTime() / MILLISECONDS_FROM_DAY)
+                .collect(Collectors.toList());
+    }
+
+    public List<Fruit> getAddedFruits(Date date, Fruit.FruitType type) {
+        return getAddedFruits(date).stream().filter(fruit -> type == fruit.getType()).collect(Collectors.toList());
+    }
+
+    public void sell(String pathToJsonFile) {
+        File file = new File(pathToJsonFile);
+        ObjectMapper mapper = new ObjectMapper();
+        List<Client> clients = null;
+
+        try {
+            clients = Arrays.asList(mapper.readValue(file, Client[].class));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        if (clients==null || fruitsDB.isEmpty())
+            return;
+
+        for (Client client : clients) {
+            int countOfFruit = 0;
+            for (Fruit fruit : fruitsDB) {
+                if (countOfFruit == client.getCount())
+                    break;
+                if (client.getType() == fruit.getType())
+                    countOfFruit++;
+            }
+            if (client.getCount() == countOfFruit)
+                pay(client.getType(), client.getCount());
```

Comment:
client.getType() - from the first point of view it seams like it's a type of client, not fruit. Need to rename